lib/api.ml

```diff
@@ -41,9 +41,17 @@ let request_headers api_user api_pwd =
     ]
 ;;
 
-type request_method = GET
+type request_method = | GET | POST
```

Comment:
I think I started this off wrong. Now that I see the second `request_method`, I don't think we want to talk about request-methods at all but construct different types of `Api_requests`. Something along the lines of 

```ocaml
type body

type 'a request_info =
  { endpoint : string
  ; request_args : request_args
  ; decoder : 'a Decoder.Yojson.Safe.decoder
  }

type 'a api_request =
  | Api_get_request of 'a request_info
  | Api_post_request of 'a request_info * body
```

That would also remove the optional body and make it explicit that `GET`-Requests never have a body.

Reply:
Addressed this with #[017e971](https://github.com/neshtea/kimai_report/pull/1/commits/017e971838ff86ab3ad87dcff90ad51cfeab48f7)